src/Mvc/Mvc.TagHelpers/src/ComponentTagHelper.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Mvc.TagHelpers
+{
+    /// <summary>
+    /// A <see cref="TagHelper"/> that renders a Razor component.
+    /// </summary>
+    [HtmlTargetElement("component", Attributes = ComponentTypeName, TagStructure = TagStructure.WithoutEndTag)]
+    public class ComponentTagHelper : TagHelper
```

Comment:
Should we seal? Would this do something reasonable if someone inherited? I don't have a strong feeling either way.

Reply:
We could do it. We've generally left all our tag helpers unsealed in the past, but that's because most of our types are unsealed by default.